src/mock_prover.rs

```diff
@@ -0,0 +1,204 @@
+use std::cell::Cell;
+
```

Comment:
It looks good. However, it's better not to put this file in root folder. what about in tests folder or in ivc/tests folder?

Reply:
We need it in the public, it shouldn't be in `tests` (because it won't be exported by default), and it has nothing to do with 'ivc'

We don't have 'tools' or anything like that, so there's no other place for dev tooling.

Reply:
The name mock_prover in root folder is kind of weird. Maybe put it in dev folder

Reply:
another option is under util.rs or utils/mock_prover.rs

Reply:
moved to util module